src/agent/test_tool/mock_server/mock_server.py

```diff
@@ -0,0 +1,208 @@
+SECRET_KEY = "my_secret_key"
+
+log_file_path = None
+
+def generate_authentication_response(expiration_seconds=60):
+    # Set the expiration time in seconds
+    expiration_time = datetime.utcnow() + timedelta(seconds=expiration_seconds)
+    expiration_timestamp = int(expiration_time.timestamp())
+
+    # Generate the JWT token with the data and the configured expiration time
+    payload = {
+        "iss": "wazuh",
+        "aud": "Wazuh Communications API",
+        "iat": datetime.utcnow(),
+        "exp": expiration_time,
+        "uuid": "edab9ef6-f02d-4a4b-baa4-f2ad12789890"
+    }
+
+    token = jwt.encode(payload, SECRET_KEY, algorithm="HS256")
+
+    response = {
+        "token": token,
+        "exp": expiration_timestamp,
+        "error": 0
+    }
+
+    return json.dumps(response)
+
+def log_request(method, endpoint, headers, body):
+    if log_file_path:
+        with open(log_file_path, 'a') as log_file:
+            timestamp = datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+            log_file.write(f"\n[{timestamp}] {method} {endpoint}\n")
+            log_file.write(f"Headers:\n{headers}\n")
+            log_file.write(f"Body:\n{body}\n\n")
+
+class MockHandler(BaseHTTPRequestHandler):
+    def _set_headers(self, code, content_type="application/json", content_length=None):
+        self.send_response(code)
+        self.send_header("Content-type", content_type)
+        if content_length:
+            self.send_header("Content-Length", str(content_length))
+        self.send_header("Connection", "close")
+        self.end_headers()
+
+    def _load_response(self, filename):
+        try:
+            with open(os.path.join(RESPONSES_DIR, filename), 'r') as f:
+                return f.read()
+        except FileNotFoundError:
+            return json.dumps({"error": "Response file not found"})
+
+    def do_POST(self):
+        content_length = int(self.headers.get('Content-Length', 0))
+        body = self.rfile.read(content_length).decode('utf-8')
+        log_request("POST", self.path, self.headers, body)
+
+        response = None
+        if self.path == "/security/user/authenticate":
+            response = self._load_response("user_authenticate.json")
+            self._set_headers(code=200, content_length=len(response))
+        elif self.path == "/agents":
+            response = self._load_response("agents.json")
+            self._set_headers(code=200, content_length=len(response))
+        elif self.path == "/api/v1/authentication":
+            response = generate_authentication_response()
+            self._set_headers(code=200, content_length=len(response))
+        elif self.path == "/api/v1/events/stateful":
+            #response = self._load_response("events_stateful.json")
+            self._set_headers(code=200) # If an response is given, add the size of the response.
+        elif self.path == "/api/v1/events/stateless":
+            #response = self._load_response("events_stateless.json")
```

Comment:
Is this commented out code necessary?

Reply:
since these endpoints publish information, for the moment it only returns code=200, in case a response is included this line must be uncommented.